### PR TITLE
[Conf/CodeClean] remove duplicated code

### PR DESCRIPTION
**Pull request by jaeyun-jung**
add internal function for sub-plugin scan.

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/nnstreamer_subplugin.c [hunk 1]
```diff
@@ -91,19 +133,11 @@ get_subplugin (subpluginType type, const char *name)
...
-      dlerror ();
-      G_UNLOCK (splock);
-      handle = dlopen (fullpath, RTLD_NOW);
-      /* If this is a correct subplugin, it will register itself */
-      G_LOCK (splock);
-      if (NULL == handle) {
-        g_critical ("Cannot dlopen %s with error %s.", fullpath, dlerror ());
-        continue;
-      }
+      _search_subplugin (type, info.names[i], info.paths[i]);
```
Comment (dongju-chae) [inline gst/nnstreamer/nnstreamer_subplugin.c]: Now no need to check whether it's successful or not?

Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_subplugin.c]: currently the result is ignored here for the full-scan and registration. actually result will be checked in line 155.

File: gst/nnstreamer/nnstreamer_subplugin.c [hunk 2]
```diff
@@ -66,24 +63,65 @@ static subpluginSearchLogic searchAlgorithm[] = {
...
+    g_critical ("Cannot dlopen %s(%s) with error %s.", name, path, dlerror ());
+    return NULL;
+  }
+
+  spdata = _get_subplugin_data (type, name);
+  if (spdata == NULL) {
+    g_critical
+        ("nnstreamer_subplugin of %s(%s) is broken. It does not call register_subplugin with its init function.",
+        name, path);
+    dlclose (handle);
```
Comment (kparichay) [inline gst/nnstreamer/nnstreamer_subplugin.c]: I think the handle should be closed outside the if statement.

Comment (myungjoo) [reply gst/nnstreamer/nnstreamer_subplugin.c]: No. the handle should be kept alive if its contents (the subplugin) are to be used elsewhere. That's why I'm working on #2049 although I'm experiencing strange behaviors with it.

**Pull request merged**